nvi-commons/src/main/java/no/sikt/nva/nvi/common/db/ApprovalStatusDao.java

```diff
@@ -0,0 +1,86 @@
+package no.sikt.nva.nvi.common.db;
+
+import static no.sikt.nva.nvi.common.DatabaseConstants.HASH_KEY;
+import static no.sikt.nva.nvi.common.DatabaseConstants.SORT_KEY;
+import static no.sikt.nva.nvi.common.DatabaseConstants.DATA_FIELD;
+import java.util.UUID;
+import no.sikt.nva.nvi.common.db.model.DbApprovalStatus;
+import nva.commons.core.JacocoGenerated;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbAttribute;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbImmutable;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbSortKey;
+
+@DynamoDbImmutable(builder = ApprovalStatusDao.Builder.class)
+public record ApprovalStatusDao(
+
+    UUID identifier,
+    @DynamoDbAttribute(DATA_FIELD) DbApprovalStatus approvalStatus
+) implements DynamoEntryWithRangeKey {
+
+    public static final String TYPE = "APPROVAL_STATUS";
+
+    public static String sk0(String institutionUri) {
+        return String.join(FIELD_DELIMITER, TYPE, institutionUri);
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    @DynamoDbPartitionKey
+    @DynamoDbAttribute(HASH_KEY)
+    public String primaryKeyHashKey() {
+        return CandidateDao.pk0(identifier.toString());
+    }
+
+    @Override
+    @DynamoDbSortKey
+    @DynamoDbAttribute(SORT_KEY)
+    public String primaryKeyRangeKey() {
+        return sk0(approvalStatus.institutionId().toString());
```

Comment:
Why not
CANDIDATE#123-123-123#APPROVAL_STATUS#1
This is the pattern we use in publication API

Reply:
Does it have any added benefit?